### PR TITLE
*: support pyroscope

### LINKED ISSUE #52318: Better Continuous Profiling
## Enhancement

Continuous profiling does not provide a good experience in TiDB; it lacks stack search capabilities and is not convenient for examining profiles over extended periods. It also does not support visualization for goroutines. Here, we can leverage the infrastructure of [Pyroscope](https://grafana.com/oss/pyroscope/) to implement this system, which will be far more reliable than what we have developed on our own.

for example, we can find this which can be optimized.

image

-----------------------------

- [x] https://github.com/pingcap/tidb/issues/52321
- [x] https://github.com/pingcap/tidb/issues/53861
- [x] https://github.com/pingcap/tidb/issues/53866
- [ ] https://github.com/pingcap/tidb/issues/53886
- [ ] https://github.com/pingcap/tidb/issues/52219

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close #52318

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)

```
export PYROSCOPE_SERVER_ADDRESS=< get from grafana cloud > 
export PYROSCOPE_BASIC_AUTH_USER=< get from grafana cloud > 
export PYROSCOPE_BASIC_AUTH_PASSWORD=< get from grafana cloud > 
bin/tidb-server --config "conf/tidb.toml" -P 4000 --store tikv --path 127.0.0.1:2379 --host 0.0.0.0 --advertise-address 0.0.0.0 -

```

- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
support pyroscope

支持 pyroscope
```

### CODE REVIEW

File: cmd/tidb-server/main.go [hunk 1]
```diff
@@ -1003,3 +1005,24 @@ func closeStmtSummary() {
 		stmtsummaryv2.Close()
 	}
 }
+
+func EnablePyroscope() {
```
Comment (lance6716) [inline cmd/tidb-server/main.go:1009]: no need to be public?

**Review by lance6716 — APPROVED** [review]

File: cmd/tidb-server/main.go [hunk 2]
```diff
@@ -1003,3 +1005,25 @@ func closeStmtSummary() {
 		stmtsummaryv2.Close()
 	}
 }
+
+func enablePyroscope() {
+	if os.Getenv("PYROSCOPE_SERVER_ADDRESS") != "" {
```
Comment (winoros) [inline cmd/tidb-server/main.go:1010]: Make it a start parameter or config so that we can integrate it with our infra in the future?

Comment (hawkingrei) [reply cmd/tidb-server/main.go:1010]: Yes， I think it needs to add this config. but this requires the PM's approval.

**Comment by hawkingrei** [comment]
/unhold 

I have communicated with PM. He agreed to merge it.

**Review by XuHuaiyu — APPROVED** [review]

**Pull request merged**